src/sentry/replays/usecases/reader.py

```diff
@@ -0,0 +1,256 @@
+from __future__ import annotations
+
+import uuid
+import zlib
+from concurrent.futures import ThreadPoolExecutor
+from datetime import datetime, timedelta
+from typing import List
+
+from snuba_sdk import (
+    Column,
+    Condition,
+    Direction,
+    Entity,
+    Granularity,
+    Limit,
+    Op,
+    OrderBy,
+    Query,
+    Request,
+)
+
+from sentry.replays.lib.storage import FilestoreBlob, RecordingSegmentStorageMeta, StorageBlob
+from sentry.replays.models import ReplayRecordingSegment
+from sentry.utils.snuba import raw_snql_query
+
+# METADATA QUERY BEHAVIOR.
+
+
+def fetch_segments_metadata(
+    project_id: int,
+    replay_id: str,
+    offset: int,
+    limit: int,
+) -> List[RecordingSegmentStorageMeta]:
+    """Return a list of recording segment storage metadata."""
+    # NOTE: This method can miss segments that were split during the deploy.  E.g. half were on
+    # filestore the other half were on direct-storage.
+
+    # TODO: Filestore is privileged until the direct storage is released to all projects.  Once
+    # direct-storage is the default driver we need to invert this.  90 days after deployment we
+    # need to remove filestore querying.
+    segments = fetch_filestore_segments_meta(project_id, replay_id, offset, limit)
+    if segments:
+        return segments
+
+    return fetch_direct_storage_segments_meta(project_id, replay_id, offset, limit)
+
+
+def fetch_segment_metadata(
+    project_id: int,
+    replay_id: str,
+    segment_id: int,
+) -> RecordingSegmentStorageMeta | None:
+    """Return a recording segment storage metadata instance."""
+    # TODO: Filestore is privileged until the direct storage is released to all projects.  Once
+    # direct-storage is the default driver we need to invert this.  90 days after deployment we
+    # need to remove filestore querying.
+    segment = fetch_filestore_segment_meta(project_id, replay_id, segment_id)
+    if segment:
+        return segment
+
+    return fetch_direct_storage_segment_meta(project_id, replay_id, segment_id)
+
+
+def fetch_filestore_segments_meta(
+    project_id: int,
+    replay_id: str,
+    offset: int,
+    limit: int,
+) -> List[RecordingSegmentStorageMeta]:
+    """Return filestore metadata derived from our Postgres table."""
+    segments: List[ReplayRecordingSegment] = (
+        ReplayRecordingSegment.objects.filter(project_id=project_id, replay_id=replay_id)
+        .order_by("segment_id")
+        .all()[offset : limit + offset]
+    )
+    return [
+        RecordingSegmentStorageMeta(
+            project_id=project_id,
+            replay_id=replay_id,
+            segment_id=segment.segment_id,
+            retention_days=0,
```

Comment:
what does `retention_days=0` semantically mean here? what do you think about having this be optional?